### ISSUE TITLE
Missing icon in standalone Enso

### ISSUE THREAD

**Issue by hubertp** [enso-org/enso#6480]
### What type of issue is this?

Transient – Occurring only once

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

Built locally with `./run ide` but I guess that shouldn't matter much.

![Screenshot from 2023-05-01 11-43-08](https://user-images.githubusercontent.com/292128/235436732-138143d9-b8c8-48bf-bfa1-418132d349f2.png)

### Expected behaviour

App should be packaged with the right icon.

### Enso Version

efe904cd9f0c44c112a254229d94522802c246e5

### Browser or standalone distribution

Standalone distribution

### Browser Version or standalone distribution

standalone

### Operating System

Linux

**Comment by 4e6** [comment]
I remember it was working previously (for pre-built packages). The electron-builder-config provides `icns` file with the icon https://github.com/enso-org/enso/blob/6b0c682b08b7745368cec3474cc774c51a5b48f5/app/ide-desktop/lib/client/electron-builder-config.ts#L120

The icon size may be an issue https://www.electron.build/icons.html
> icon.icns (macOS app icon) or icon.png. Icon size should be at least 512x512.

I can see that on Linux the icon has 64x64 resolution
```bash
$ file (readlink enso.png)
usr/share/icons/hicolor/32x32/apps/enso.png: PNG image data, 64 x 64, 8-bit/color RGBA, non-interlaced
```

I'm not sure how to check the `icns` file on macOS.

@hubertp did it work for you previously, e.g. when you install nightlies, or you have never seen the Enso icon? Did you use `./run ide build` or `./run ide start` command? 

@mwu-tow can it be the case that the local `./run ide` execution does not fetch the icon file?

<blockquote><div><strong><a href="https://www.electron.build/icons.html">Icons - electron-builder</a></strong></div></blockquote>

**Comment by hubertp** [comment]
I believe it used to work for me and I had the icon correctly (so sounds like a regression).
I just tried with the latest nightly (via AppImage) and it shows the same thing as when built locally, making the bug rather high priority.

Locally I built using `./run ide start`.

**Comment by mwu-tow** [comment]
> @mwu-tow can it be the case that the local ./run ide execution does not fetch the icon file?

I don't think so, icon is generated locally. I'd expect that if it was missing, the electron-builder should raise an error (though I don't know for sure).

@hubertp 
> Locally I built using `./run ide start`.

This won't give you an icon. The `ide start` does not packages the Electron application but just runs it using the development mode, so no package is being created and no icon will be present.

**Comment by hubertp** [comment]
> > @mwu-tow can it be the case that the local ./run ide execution does not fetch the icon file?
> 
> I don't think so, icon is generated locally. I'd expect that if it was missing, the electron-builder should raise an error (though I don't know for sure).
> 
> @hubertp
> 
> > Locally I built using `./run ide start`.
> 
> This won't give you an icon. The `ide start` does not packages the Electron application but just runs it using the development mode, so no package is being created and no icon will be present.

OK, but executing via AppImage it should work, right?

**Comment by mwu-tow** [comment]
> > > @mwu-tow can it be the case that the local ./run ide execution does not fetch the icon file?
> > 
> > I don't think so, icon is generated locally. I'd expect that if it was missing, the electron-builder should raise an error (though I don't know for sure).
> > 
> > @hubertp
> > 
> > > Locally I built using `./run ide start`.
> > 
> > This won't give you an icon. The `ide start` does not packages the Electron application but just runs it using the development mode, so no package is being created and no icon will be present.
> 
> OK, but executing via AppImage it should work, right?

Yes, it should work.

**Comment by wdanilo** [comment]
I believe that our icon generation scripts fail if they can't generate icons. The electron builder should fail as well ... This is strange. I understand it did not happen with other packages? If so, I'd close it for now and re-open if it happens again.

**Comment by vitvakatu** [comment]
Two years later, we still have this issue, I just confirmed it on Ubuntu 25. Something weird is happening, but I should be able to investigate as part of Bazel related work in the nearest future.

**Comment by hubertp** [comment]
I think this is a won't fix with AppImage. 

The current electron-build uses .icns (Mac format) which "in theory", at east according to documentation, should work. But if you look at the extracted AppImage contents, the icon is located in /usr/share/... directory with a `0x0` subdirectory, meaning it was unable to generate a png with appropriate resolutions (16x16, 32x32,64x64, etc.)That can be easily fixed by providing the path to a directory with all scaled `png` images. Unfortunately it does not seem to fix the problem.
Neither does setting `icon` property in `electron`'s `BrowserWindow` as it is suggested for Linux in various tickets of AppImage. 
I also read that something has changed wrt to .desktop configuration on Wayland so that could be another reason why at least I couldn't get it to work. Maybe it works on x11. I also noticed that other mature AppImages products show a default icon on my machine. So all in all, it feels like a can of worms not worth pursuing atm at least.

**Comment by vitvakatu** [comment]
I will close it if no progress is made within the next two months. I still hope to address the issue.

**Comment by farmaazon** [comment]
from https://github.com/enso-org/enso/issues/14925#issuecomment-4160091532 
> - according to [this section](https://www.electron.build/icons.html#linux) the icons should work
> - stack overflow: "[there are multiple options](https://stackoverflow.com/a/76779051/2960975)"
> - FYI: https://github.com/electron-userland/electron-builder/issues/2577
> - this one is quite old: https://github.com/electron/electron/issues/6205 but somehow promissing
